pkg/runtime/core/trainingruntime.go

```diff
@@ -187,24 +187,51 @@ func (r *TrainingRuntime) mergePodSpecOverrides(trainJob *trainer.TrainJob, jobS
 			}) {
 				continue
 			}
-			patch, err := json.Marshal(podSpecOverride)
+
+			overrideBytes, err := json.Marshal(podSpecOverride)
 			if err != nil {
 				return err
 			}
-			source, err := json.Marshal(job.Template.Spec.Template.Spec)
+			overrideMap := map[string]any{}
+			if err := json.Unmarshal(overrideBytes, &overrideMap); err != nil {
+				return err
+			}
+
+			// Remove targetJobs as it's not part of the PodTemplateSpec
+			delete(overrideMap, "targetJobs")
+
+			podTemplatePatch := map[string]any{}
+
+			md := map[string]any{}
+			if labels, ok := overrideMap["labels"]; ok {
+				md["labels"] = labels
+				delete(overrideMap, "labels")
+			}
+			if annotations, ok := overrideMap["annotations"]; ok {
+				md["annotations"] = annotations
+				delete(overrideMap, "annotations")
+			}
+			podTemplatePatch["metadata"] = md
```

Comment:
Do we need to delete those values, since we didn't delete the `targetJob` from the map before: https://github.com/kubeflow/trainer/blob/master/pkg/runtime/core/trainingruntime.go#L190 ?
I think, if those values are not part of `.metadata` or `.spec` API, they will just be ignored.

Reply:
Thanks, I tested it and confirmed that not deleting those values indeed causes no issues.